com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/RenderPass/CustomPass/CustomPass.cs

```diff
@@ -172,12 +172,28 @@ internal void ExecuteInternal(ScriptableRenderContext renderContext, CommandBuff
                 {
                     Setup(renderContext, cmd);
                     isSetup = true;
+                    userMaterialPropertyBlock = new MaterialPropertyBlock();
                 }
 
                 SetCustomPassTarget(cmd);
 
+                // Create the custom pass context:
+                bool msaa = IsMSAAEnabled(hdCamera);
+                CustomPassContext ctx = new CustomPassContext(
+                    renderContext, cmd, hdCamera,
+                    cullingResult, msaa ? targets.cameraColorMSAABuffer : targets.cameraColorBuffer,
+                    rtManager.GetDepthStencilBuffer(msaa),
+                    rtManager.GetNormalBuffer(msaa),
+                    targets.customColorBuffer,
+                    targets.customDepthBuffer,
+                    userMaterialPropertyBlock
+                );
+
                 isExecuting = true;
+#pragma warning disable CS0618 // Member is obsolete
```

Comment:
Can we make the base Execute(CustomPassContext ctx) call the deprecated one instead? This way it works for old code as well but we don't have to call two things in the main path. Besides, if users forget to remove the old implementation, and still add a new one, it won't be called twice.

Reply:
Good catch :) I fixed it